Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/JsonSchemaEnhancer.ts

```diff
@@ -258,7 +260,12 @@ function schemaPropertyForNonReference(property: EntityProperty): SchemaProperty
       return { type: 'string', format: 'time', description };
 
     case 'schoolYearEnumeration':
-      return { type: 'integer', description, minimum: beginYear, maximum: endYear };
+      if (property.parentEntity.type === 'common') {
+        // For a common, the school year ends up being nested under a reference object
+        return schoolYearEnumerationSchema;
+      }
+
+      return schoolYear;
```

Comment:
This is the key change for the original problem. For a common, we end with something like

```
"cohortYears": [
    {
      "schoolYearTypeReference": { "schoolYear": 2022 },
      "cohortYearTypeDescriptor": "uri://ed-fi.org/CohortYearTypeDescriptor#Twelfth grade"
    }
```

But in other cases when we're in this function, we have a flatter structure like this:

```
"gradingPeriods": [
    {
      "gradingPeriodReference": {
        ...
        "schoolYear": 0
        }
      }
```

Note how `schoolYear` is flat here, instead of an object. So we need to support both possibilities.